app/controllers/friend_requests_controller.rb

```diff
@@ -0,0 +1,51 @@
+class FriendRequestsController < ApplicationController
+  before_action :set_friend_request, only: [:update, :destroy]
+
+
+  def index
+    @sent_requests = current_user.friend_requests_sent.pending
+    @received_requests = current_user.friend_requests_received.pending
+  end
+
+  def create
+    @friend_request = FriendRequest.find_or_initialize_by(sender_id: current_user.id, receiver_id: params[:friend_request][:receiver_id])
+    @friend_request.status = 0
+
+    if @friend_request.save
+      flash[:success] = "Friend request sent"
+    else
+      flash[:danger] = "Something went wrong: friend request not sent"
+    end
+    redirect_to users_path
+
+  end
+
+  def update
+    # @friend_request.accept
+      if @friend_request.update(status: 1)
+        @friend_request.sender.establish_friendship(@friend_request.receiver)
+        flash[:success]= "You're now friend with #{@friend_request.sender.first_name}"
+      else
+        flash[:danger] = "Something went wrong"
+      end
+      redirect_to root_path
+  end
+
+  def decline
```

Comment:
Let's not have any non-CRUD actions listed as a public method in our resource controllers. I made mention of this too in the `users_controller.rb`.